### PR TITLE
[MPS] Fix sort returning out-of-bounds indices for bool/int-max/NaN inputs

### PR DESCRIPTION
**Pull request by Isalia20**
Bitonic sort pads to power-of-2 with the dtype extremum (NaN/INT_MAX/True). Real values(from the array we give to sort) that equal the padding tied in the comparator, letting padding's out-of-range indices leak into the output.

This PR tags padding slots with a sentinel index (~IdxT(0), out of any real input's range) and the comparator  then prefers real slots over padding on value ties.

Reproducer:
```
import torch

cases = [
    ("bool",   torch.tensor([True, False, True, False, True], device="mps")),
    ("float-nan", torch.tensor([1.0, float('nan'), 2.0, float('nan'), 3.0], device="mps")),
    ("int-max", torch.tensor([1, torch.iinfo(torch.int32).max, 2], dtype=torch.int32, device="mps")),
]
for name, t in cases:
    _, idx = torch.sort(t)
    print(f"{name:12} idx={idx.tolist()}  max={idx.max().item()}  size={t.numel()}")
```

Performance, small regression:
## float32
| case | shape | before-fix (ms) | after-fix (ms) | after/before |
|---|---|---:|---:|---:|
| b1_sort4k | 1x4096 | 0.0290 | 0.0311 | 1.075x |
| b1_sort8k | 1x8192 | 0.0358 | 0.0381 | 1.062x |
| b1_sort16k | 1x16384 | 0.0565 | 0.0596 | 1.056x |
| b1_vocab32k | 1x32000 | 0.0659 | 0.0694 | 1.052x |
| b4_vocab32k | 4x32000 | 0.1103 | 0.1172 | 1.062x |

## bf16 (single-block merge path)

| case | shape | before-fix (ms) | after-fix (ms) | after/before |
|---|---|---:|---:|---:|
| bf16m_8x1024 | 8x1024 | 0.0237 | 0.0260 | 1.097x |
| bf16m_256x2048 | 256x2048 | 0.2499 | 0.2823 | 1.130x |
| bf16m_16x4096 | 16x4096 | 0.0421 | 0.0455 | 1.081x |

different shapes because the above shapes(float32 ones) dispatch to radix for bfloat16

### CODE REVIEW

**Review by malfet — APPROVED** [review]
LGTM, but can you run some perf benchmarks on this one (as new `bitonic_lt` is much slower than old one, isn't it?)

**Comment by malfet** [comment]
@claude do an adversarial review, and try to come up with value that match those "sentinels"...
